src/m_menu.c

```diff
@@ -2849,30 +2844,57 @@ setup_menu_t keys_settings2[] =  // Key Binding screen strings
 
 };
 
+static const char *layout_strings[] = {
+  "Default", "Swap", "Legacy", "Legacy Swap", NULL
+};
+
+static const char *curve_strings[] = {
+  "Linear", "1.1", "1.2", "1.3", "1.4", "1.5", "1.6", "1.7", "1.8", "1.9",
+  "Squared", "2.1", "2.2", "2.3", "2.4", "2.5", "2.6", "2.7", "2.8", "2.9",
+  "Cubed", NULL
+};
+
+#define GP_X 136
+
 setup_menu_t keys_settings3[] =
 {
-  {"GAMEPAD", S_SKIP|S_TITLE,m_null,KB_X,M_Y},
+  {"Gamepad", S_SKIP|S_TITLE, m_null, GP_X, M_Y},
+
+  {"Enable", S_YESNO, m_scrn, GP_X, M_Y + 1 * M_SPC,
+   {"joy_enable"}, 0, I_ResetController},
```

Comment:
Do we need this option? MBF has a "enable mouse" option that we removed. I think this option is similar.

Reply:
I figure it's nice to disable for a small performance improvement, but we can remove it from the menu at least.

Reply:
Removed from the menu, but left in the config for now: https://github.com/fabiangreffrath/woof/pull/1394/commits/55b544e0ef8f856e76cee3a370f4c63e37f92a60

Reply:
Having a mouse (and using it for gaming) is probably the most common environment today. I, for example, do not even own a gamepad. 

Is it possible to detect if a gamepad is connected and skip the functions otherwise? It would be nice to skip some cycles for reading from hardware that's not present.

Reply:
> Is it possible to detect if a gamepad is connected and skip the functions otherwise? It would be nice to skip some cycles for reading from hardware that's not present.

Sure, we can check `controller != NULL` and set a boolean or something similar. That sounds like a good idea.

Reply:
Fixed by https://github.com/fabiangreffrath/woof/pull/1394/commits/0a7661ef751ae242d6cbbb60a545f3d3165aa74f

Reply:
Thank.